src/main/java/baritone/utils/NotificationHelper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils;
+
+import java.awt.*;
+import java.io.IOException;
+
+public class NotificationHelper {
+    public static void notify(String text, boolean error) {
+        if (System.getProperty("os.name").contains("Linux"))
+            linux(text);
+        else
+            notification(text, error);
+    }
+
+    public static void notification(String text, boolean error) {
+        if (SystemTray.isSupported()) {
+            try {
+                SystemTray tray = SystemTray.getSystemTray();
+                Image image = Toolkit.getDefaultToolkit().createImage("");
+                // Replace with some logo
+
+                TrayIcon trayIcon = new TrayIcon(image, "Baritone");
+                trayIcon.setImageAutoSize(true);
+                trayIcon.setToolTip("Baritone");
+                tray.add(trayIcon);
+
+                if(error)
+                    trayIcon.displayMessage("Baritone", text, TrayIcon.MessageType.ERROR);
+                else
+                    trayIcon.displayMessage("Baritone", text, TrayIcon.MessageType.INFO);
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        else {
+            System.out.println("SystemTray is not supported");
+        }
+    }
+
+    /*
+     * The only way to display notifications on linux is to use the java-gnome library, or send notify-send to shell with a ProcessBuilder
+     * Unfortunately the java-gnome library is licenced under the GPL, see: (https://en.wikipedia.org/wiki/Java-gnome)
+     */
```

Comment:
baritone is also GPL, you're free to use the library here

Reply:
baritone is lgpl 

BUT

i'm like 90% sure that using a GPL utility through bash is allowed. like otherwise how could you interface with GNU or linux at all, from a proprietary program. we just can't import it as a library or use its code in any way.

Reply:
oh well you can use imported GPL in LGPL

i thought this was GPL my bad

Reply:
no, you can use imported LGPL in GPL. GPL is more restrictive, not less.